### PR TITLE
Add POM files

### PR DESCRIPTION
**Pull request by leomartinez2020**
In this PR the page object pattern is implemented

### CODE REVIEW

File: cypress/e2e/buy-shirt.cy.ts [hunk 1]
```diff
@@ -1,20 +1,49 @@
...
-    cy.get("#SubmitLogin").click();
-
-    cy.get(".cart_navigation").contains("Proceed to checkout").click();
-    cy.get("#cgv").check();
-    cy.get(".cart_navigation").contains("Proceed to checkout").click();
-    cy.get(".bankwire").click();
+    menuContentPage.visitMenuContentPage();
+    menuContentPage.goToTShirtMenu();
+
+    // Add chosen product to cart from product list
```
Comment (holgiosalos) [inline cypress/e2e/buy-shirt.cy.ts]: as the code is self explanatory, comments are not needed in this context

File: cypress/e2e/buy-shirt.cy.ts [hunk 2]
```diff
@@ -1,20 +1,49 @@
...
+    addresStepPage.checkout();
+
+    // Shipping: Agree to terms and proceed
+    shippingStepPage.checkout();
+
+    // Payment by bank wire
+    paymentStepPage.payByBank();
+
+    // Confirm order
     cy.get("#cart_navigation > .button").click();
```
Comment (holgiosalos) [inline cypress/e2e/buy-shirt.cy.ts]: this step should be part of the paymentStepPage

File: cypress/e2e/buy-shirt.cy.ts [hunk 3]
```diff
@@ -1,20 +1,49 @@
...
-    cy.get("#center_column > div > p > strong").should("have.text", "Your order on My Store is complete.");
+
+    // Assert completed order message on page
+    cy.get("#center_column > div.box > p > strong").should("have.text", "Your order on My Store is complete.");
```
Comment (holgiosalos) [inline cypress/e2e/buy-shirt.cy.ts]: this also can be a method of the paymentStepPage. You can create a method called getConfirmationMessage with the part of the cy.get. And the should part can remain in this test file

**Comment by leomartinez2020** [comment]
@holgiosalos I improved the code although I am still trying to understand the POM pattern which is new to me

File: cypress/page/payment-step.page.ts [hunk 4]
```diff
@@ -1,12 +1,19 @@
...
+        this.cartNav = "#cart_navigation > .button";
     }
 
     public payByBank() {
         cy.get(this.bankWireTag).click();
+        cy.get(this.cartNav).click();
+    }
+
+    public getConfirmationMessage() {
+        cy.get("#center_column > div.box > p > strong").should("have.text", "Your order on My Store is complete.");
```
Comment (holgiosalos) [inline cypress/page/payment-step.page.ts]: store the locator in a variable (I would call it confirmationMessageText, but you can choose another name), also, make this method to return the element with the title:
`return cy.get(this.confirmationMessageText);`

The assertion (`.should("have.text", "Your order on My Store is complete.");`) should go on the buy-tshirt.cy.ts file

File: cypress/e2e/buy-shirt.cy.ts [hunk 5]
```diff
@@ -21,29 +21,18 @@ describe("Buy a t-shirt", () => {
...
 
-    // Payment by bank wire
     paymentStepPage.payByBank();
-
-    // Confirm order
-    cy.get("#cart_navigation > .button").click();
-
-    // Assert completed order message on page
-    cy.get("#center_column > div.box > p > strong").should("have.text", "Your order on My Store is complete.");
+    paymentStepPage.getConfirmationMessage();
```
Comment (holgiosalos) [inline cypress/e2e/buy-shirt.cy.ts]: Here I would expect to have the assertion:
`paymentStepPage.getConfirmationMessage().should("have.text", "Your order on My Store is complete.");`

**Comment by leomartinez2020** [comment]
@holgiosalos I think corrected the code and maybe I have it right

File: cypress/e2e/buy-shirt.cy.ts [hunk 6]
```diff
@@ -33,6 +34,7 @@ describe("Buy a t-shirt", () => {
     shippingStepPage.checkout();
 
     paymentStepPage.payByBank();
-    paymentStepPage.getConfirmationMessage();
+    const confirmationMessage = paymentStepPage.getConfirmationMessage();
+    confirmationMessage.should("have.text", expectedText);
```
Comment (holgiosalos) [inline cypress/e2e/buy-shirt.cy.ts]: there is no need to create the const confirmationMessage
```suggestion
    paymentStepPage.getConfirmationMessage()
        .should("have.text", expectedText);
```

File: cypress/page/payment-step.page.ts [hunk 7]
```diff
@@ -13,7 +16,8 @@ class PaymentStepPage {
     }
 
     public getConfirmationMessage() {
-        cy.get("#center_column > div.box > p > strong").should("have.text", "Your order on My Store is complete.");
+        this.confirmationMessageText = cy.get(this.confirmationMessageTag);
+        return this.confirmationMessageText;
```
Comment (holgiosalos) [inline cypress/page/payment-step.page.ts]: there is no need of the confirmationMessageText property
```suggestion
        return cy.get(this.confirmationMessageTag);
```

**Comment by leomartinez2020** [comment]
@holgiosalos hopefully this time the code is cleaner

**Review by holgiosalos — APPROVED** [review]

**Pull request merged**